docs/DeveloperGuide.md

```diff
@@ -1,34 +1,662 @@
+ the same line, separated by "|". Nothing will be changed in `MemberList`.  
+ 
+ The following shortcut commands can achieve the same result: <br/>
+ `hr list`<br/>
+ `hr l`<br/>
+ 
+ The sequence diagram for listing the members is as shown below:
+ 
+ ![CommandViewMember](hrDiagramPic/CommandViewMember.png)
+
+**3.5.3. Change member information**  
+3.5.3.1. Current Implementation  
+The `CommandChangeInfo` class in `seedu.duke.hr` handles changing contacts and roles information of the members in 
+`MemberList` and showing the contacts and roles of the changed `Member`.  
+It implements the following operation:  
+* `CommandChangeInfo#execute()` - Changes any of the their contacts and roles `Member` in `MemberList` and shows the 
+modified member information.  
+
+Given below is an example usage scenario and how the program list the information of members.  
+
+Step 1. After some `hr addMember` commands, the user created a `MemberList` with two `Member`. <br/>
+The first `Member` is "John Sterling" with phone number "12345678", email "123@gmail.com", role "member". <br/>
+The second `Member` is "Harry Potter", phone number "88888888", email "qaz@gmail.com", role "president". <br/>
+
+Step 2. The user executes `hr changeInfo /n john sterling /p 11111111 /r publicity director` command to change the phone
+ number and role of the member with name "John Sterling" in the list. The `hr changeInfo` command calls 
+`CommandChangeInfo#execute()`, then `Member` with the `memberName` `John Sterling` in the `MemberList` will have its 
+`memberPhone` changed to `11111111`, and `memberRole` changed to `publicity director`. The `memberName` is not case 
+sensitive. 
+
+The following shortcut commands can achieve the same result: <br/>
+`hr c /n john Sterling /p 11111111 /r publicity director`<br/>
+
+The sequence diagram for changing contacts and role information of a member is as shown below:
+
+![CommandChangeMemberInfo](hrDiagramPic/CommandChangeMemberInfo.png)
+
+**3.5.4. Design Considerations**  
+Aspect: Changing member information 
+*Alternative 1(Current Choice): Member information is to be modified based on the member's full name.  
+    *Pros: Easy to implement. Also, if the user knows the name of the target member, which is a likely case in actual 
+    practice, he can change the member's information quickly.
+    *Cons: Member name cannot be easily modified. If the user wants to change the name of the member, the user has to delete 
+    the target member, and add the member back using the new name.
+*Alternative 2: Member Information is to be modified based on the member's index in the list.  
+    *Pros: Member name can be easily modified. 
+    *Cons: This feature is very dependent on the list member feature. The user will always need to call the `hr listMember` 
+    command to find out the index of the target member, before he can change the member's information.  
+
+### 3.6. Storage
+
+![](BackendDiagram/StorageFlow.png)
```

Comment:
<img width="309" alt="Screenshot 2020-10-30 at 12 34 55" src="https://user-images.githubusercontent.com/59906930/97660383-5c4a4380-1aac-11eb-8966-61064f8b8a84.png">

The loop has no conditions. Is this missed out intentionally?